bert_e/git_host/github/__init__.py

```diff
@@ -528,30 +550,79 @@ def sort_f(check_run):
                 return datetime.strptime(date, '%Y-%m-%dT%H:%M:%SZ')
             return datetime(year=1962, month=1, day=1)
 
-        self.data['check_runs'].sort(key=sort_f, reverse=True)
+        self.data['check_suites'].sort(key=sort_f, reverse=True)
 
-        failed_c = [elem for elem in self.data['check_runs']
+        failed_c = [elem for elem in self.data['check_suites']
                     if elem['conclusion'] != 'success']
         if len(failed_c):
             return failed_c[0]['html_url']
-        elif len(failed_c) and len(self.data['check_runs']):
-            return self.data['check_runs'][0]['html_url']
+        elif len(failed_c) and len(self.data['check_suites']):
+            return self.data['check_suites'][0]['html_url']
         return ''
 
+    def is_pending(self):
+        return len([
+            elem for elem in self._check_suites if elem['status'] == 'pending'
+        ]) > 0
+
+    def _get_check_suite_ids(self, workflow_runs):
+        return list(
+            map(lambda elem: elem['check_suite_id'], workflow_runs)
+        )
+
+    def _get_aggregate_workflow_dispatched(self):
+        ref = self._check_suites[0]['head_sha']
+        repo = self._check_suites[0]['repository']['name']
+        owner = self._check_suites[0]['repository']['owner']['login']
+
+        return AggregatedWorkflowRuns.get(
+            client=self.client,
+            owner=owner, repo=repo, ref=ref,
+            params={
+                'event': 'workflow_dispatch',
+                'branch': self._check_suites[0]['head_branch']
+            })
+
+    def remove_unwanted_workflow_dispatch(self):
```

Comment:
I would rename this one as it's not only the workflow dispatch
```suggestion
    def remove_unwanted_workflows(self):
```